src/main/resources/db/migration/V2__create_item_cat_auction.sql

```diff
@@ -0,0 +1,43 @@
+BEGIN;
+CREATE TABLE IF NOT EXISTS category (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  name text NOT NULL,
+  subcategory UUID,
+  CONSTRAINT fk_subcategory
+      FOREIGN KEY(subcategory) 
+	  REFERENCES category(id)
+);
+
+CREATE TABLE IF NOT EXISTS item (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  name text NOT NULL,
+  start_price decimal,
+  color text,
+  size numeric,
+  description text NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS auction (
+  id UUID DEFAULT gen_random_uuid () PRIMARY KEY,
+  start_date timestamp,
+  end_date timestamp,
+  highest_bid decimal,
+  address text,
+  zip_code numeric,
+  phone text,
+  status text,
+  shipping_cost_included text,
+  item_id UUID,
+  seller_id UUID,
+  category_id UUID,
```

Comment:
The category should be tied to an item, not an auction.

Reply:
I was thinking about that. Initially, I tied it to the item table. But when I was working on the landing page, I realized records of items will never be displayed on the platform as such, but rather records of auctions that represent those items will be the ones for display. So then I tied auctions with category to not have to do constant querying over the table of items to fetch the required category as we actually need to categorize auctions on the platform.

What do you think, should I still change it back to the item table?